src/Orchard.Cms.Web/Modules/Orchard.OpenId/Controllers/AdminController.cs

```diff
@@ -223,11 +189,20 @@ public class AdminController : Controller
                 AllowHybridFlow = model.AllowHybridFlow
             };
 
-            await _applicationManager.CreateAsync(openIdApp);
+            if (model.Type == ClientType.Confidential)
+            {
+                await _applicationManager.CreateAsync(application, model.ClientSecret, HttpContext.RequestAborted);
```

Comment:
Hum, good point. I guess we could use the `Enum.Parse` overload accepting an `ignoreCase` parameter.